frontend/src/metabase/containers/ItemPicker/ItemPicker.tsx

```diff
@@ -33,6 +33,7 @@ interface OwnProps<TId> {
   entity?: typeof Collections; // collections/snippets entity
   showSearch?: boolean;
   showScroll?: boolean;
+  showOnlyPersonalCollections?: boolean;
```

Comment:
I'm not sure we'll need that yet. `ItemPicker` is only used in `DashboardPicker` (Query builder) while `QuestionPicker` (dashboards) uses totally different components.

We might need to come up with something else when dealing with [hiding personal collections/dashboards/and questions when configuring click behavior](https://metaboat.slack.com/archives/C057T1QTB3L/p1698664681662569?thread_ts=1698664221.417809&cid=C057T1QTB3L) though since that uses `ItemPicker`.

What about we leave it like this and when we're about to work on the click behavior we could change it to something similar [on the BE](https://github.com/metabase/metabase/pull/34887)?

The BE search endpoint now accepts `filter_items_in_personal_collection` with either `exclude` or `only`. In FE's case, it's like an enum with 3 states. `null`, `exclude`, `only` where
- `null`: renders everything
- `exclude`: renders only items in public collections
- `only`: renders only items in personal collections

What do you think?

Reply:
Good point about dashboards using a different component. Feel free to leave it as it is for now